xbmc/utils/Job.h

```diff
@@ -64,6 +65,19 @@ class IJobCallback
    \sa CJobManager and CJob
    */
   virtual void OnJobProgress(unsigned int jobID, unsigned int progress, unsigned int total, const CJob *job) {};
+
+  /*!
+   \brief An optional callback function used when a job will be aborted before it was executed.
+
+   OnJobAbort is called whenever a job gets aborted before it was executed. Job's DoWork method
```

Comment:
Would this ever be used in the case that a running job needs to be aborted?

Reply:
No, if a running job shall be canceled, this can be done via OnJobProgress.

Reply:
Thanks for the explanation.

Reply:
> Would this ever be used in the case that a running job needs to be aborted?

 On second thought, you are right. This new callback has also be called when job manager cancels a running job. I will change the PR accordingly.